frappe/printing/doctype/letter_head/letter_head.py

```diff
@@ -78,12 +76,15 @@ def set_image_as_html(
 		frappe.msgprint(success_msg, alert=True)
 
 	def on_update(self):
-		self.set_as_default()
+		self.update_system_defaults()
+
+	def on_trash(self):
+		if self.is_default:
+			frappe.throw(_("Default Letter Head cannot be deleted"))
 
-		# clear the cache so that the new letter head is uploaded
-		frappe.clear_cache()
+		self.update_system_defaults()
```

Comment:
If `is_default` was always updated through the controller, this would not be required. But if `is_default` was unset in the database only, this is required to make sure the defaults are cleared.

Reply:
this looks bad to me tbh

suppose there are 2 letter heads and one of them is default and the non-default one is deleted, it will clear the defaults (why?)

Reply:
removed default_letter_head_content